docs/en/advanced_tutorials/data_element.md

```diff
@@ -1,3 +1,1096 @@
+
+```python
+import itertools
+
+class TmpObject:
+    def __init__(self, tmp) -> None:
+        assert isinstance(tmp, list)
+        self.tmp = tmp
+
+    def __len__(self):
+        return len(self.tmp)
+
+    def __getitem__(self, item):
+        if type(item) == int:
+            if item >= len(self) or item < -len(self):  # type:ignore
+                raise IndexError(f'Index {item} out of range!')
+            else:
+                # keep the dimension
+                item = slice(item, None, len(self))
+        return TmpObject(self.tmp[item])
+
+    @staticmethod
+    def cat(tmp_objs):
+        assert all(isinstance(results, TmpObject) for results in tmp_objs)
+        if len(tmp_objs) == 1:
+            return tmp_objs[0]
+        tmp_list = [tmp_obj.tmp for tmp_obj in tmp_objs]
+        tmp_list = list(itertools.chain(*tmp_list))
+        new_data = TmpObject(tmp_list)
+        return new_data
+
+    def __repr__(self):
+        return str(self.tmp)
+```
+
+```python
+img_meta = dict(img_shape=(800, 1196, 3), pad_shape=(800, 1216, 3))
+instance_data = InstanceData(metainfo=img_meta)
+instance_data.det_labels = torch.LongTensor([2, 3])
+instance_data["det_scores"] = torch.Tensor([0.8, 0.7])
+instance_data.bboxes = torch.rand((2, 4))
+instance_data.polygons = TmpObject([[1, 2, 3, 4], [5, 6, 7, 8]])
+print(instance_data)
+```
+
+```
+<InstanceData(
+
+    META INFORMATION
+    pad_shape: (800, 1216, 3)
+    img_shape: (800, 1196, 3)
+
+    DATA FIELDS
+    det_labels: tensor([2, 3])
+    polygons: [[1, 2, 3, 4], [5, 6, 7, 8]]
+    det_scores: tensor([0.8000, 0.7000])
+    bboxes: tensor([[0.4207, 0.0778, 0.9959, 0.1967],
+                [0.4679, 0.7934, 0.5372, 0.4655]])
+) at 0x7fa061b5d2b0>
+```
+
+```python
+# advanced indexing
+print(instance_data[instance_data.det_scores > 0.75])
+```
+
+```
+<InstanceData(
+
+    META INFORMATION
+    pad_shape: (800, 1216, 3)
+    img_shape: (800, 1196, 3)
+
+    DATA FIELDS
+    bboxes: tensor([[0.4207, 0.0778, 0.9959, 0.1967]])
+    det_labels: tensor([2])
+    det_scores: tensor([0.8000])
+    polygons: [[1, 2, 3, 4]]
+) at 0x7f9f312716d0>
+```
+
+```python
+# cat
+print(InstanceData.cat([instance_data, instance_data]))
+```
+
+```
+<InstanceData(
+
+    META INFORMATION
+    pad_shape: (800, 1216, 3)
+    img_shape: (800, 1196, 3)
+
+    DATA FIELDS
+    bboxes: tensor([[0.4207, 0.0778, 0.9959, 0.1967],
+                [0.4679, 0.7934, 0.5372, 0.4655],
+                [0.4207, 0.0778, 0.9959, 0.1967],
+                [0.4679, 0.7934, 0.5372, 0.4655]])
+    det_labels: tensor([2, 3, 2, 3])
+    det_scores: tensor([0.8000, 0.7000, 0.8000, 0.7000])
+    polygons: [[1, 2, 3, 4], [5, 6, 7, 8], [1, 2, 3, 4], [5, 6, 7, 8]]
+) at 0x7f9f31271490>
+```
+
+### PixelData
+
+[`PixelData`](mmengine.structures.PixelData) builds on `BaseDataElement` which also has restrictions on the stored data.
+
+- All data must be in three-dimension in the order of (Channel, Height, Width).
```

Comment:
Conflicts with the description [here](https://github.com/open-mmlab/mmengine/pull/1067/files#diff-f82d5149d3224e872d55307e9b68092e5526421d2c82c4e6cfdf793ad4c94235R378)

Please check whether the data must be three-dimension

Reply:
> Conflicts with the description [here](https://github.com/open-mmlab/mmengine/pull/1067/files#diff-f82d5149d3224e872d55307e9b68092e5526421d2c82c4e6cfdf793ad4c94235R378)
> 
> Please check whether the data must be three-dimension

`PixelData` is three-dimensional, and I believe the linked content is talking about the data that can be encapsulated with `PixelData`. So this should be fine.